styx-scheduler-service/src/main/java/com/spotify/styx/Scheduler.java

```diff
@@ -103,14 +114,17 @@ public Scheduler(Time time, TimeoutConfig ttls, StateManager stateManager,
     this.resourceDecorator = Objects.requireNonNull(resourceDecorator);
     this.stats = Objects.requireNonNull(stats);
     this.dequeueRateLimiter = Objects.requireNonNull(dequeueRateLimiter, "dequeueRateLimiter");
+    this.gate = Objects.requireNonNull(gate, "gate");;
   }
 
   void tick() {
     final Map<String, Resource> resources;
     final Optional<Long> globalConcurrency;
+    final StyxConfig config;
```

Comment:
this part needs a bit of rework if we merge https://github.com/spotify/styx/pull/360

Reply:
I think we can hold off on #360 for now.

Reply:
👍